chain/chain/src/state_sync/utils.rs

```diff
@@ -203,30 +206,31 @@ pub(crate) fn update_sync_hashes<T: ChainStoreAccess>(
     on_new_header(chain_store, store_update, header)
 }
 
-///. Returns whether `block_hash` is the block that will appear immediately before the "sync_hash" block. That is,
-/// whether it is going to be the prev_hash of the "sync_hash" block, when it is found.
+/// Returns whether `tip.last_block_hash` is the block that will appear immediately before the "sync_hash" block.
+/// That is, whether it is going to be the prev_hash of the "sync_hash" block, when it is found.
 ///
-/// `block_hash` is the prev_hash of the future "sync_hash" block iff it is the first block for which the
+/// `tip.last_block_hash` is the prev_hash of the future "sync_hash" block iff it is the first block for which the
 /// number of new chunks in the epoch in each shard is at least 2
 ///
 /// This function can only return true before we save the "sync_hash" block to the `StateSyncHashes` column,
 /// because it relies on data stored in the `StateSyncNewChunks` column, which is cleaned up after that.
 ///
 /// This is used when making state snapshots, because in that case we don't need to wait for the "sync_hash"
 /// block to be finalized to take a snapshot of the state as of its prev prev block
-pub(crate) fn is_sync_prev_hash(
-    store: &Store,
-    block_hash: &CryptoHash,
-    prev_hash: &CryptoHash,
-) -> Result<bool, Error> {
-    let Some(new_chunks) = get_state_sync_new_chunks(store, block_hash)? else {
+pub(crate) fn is_sync_prev_hash(chain_store: &ChainStoreAdapter, tip: &Tip) -> Result<bool, Error> {
+    if let Some(sync_hash) = chain_store.get_current_epoch_sync_hash(&tip.epoch_id)? {
```

Comment:
Can you add a short comment about why this is important? Otherwise it looks like just an optimization but it's actually about correctness.